### PR TITLE
Viewer: Show dropzone when no reviewables

### LINKED ISSUE #639: Viewer: Show dropzone when no reviewables
When there are no reviewables show the dropzone to upload new reviewables, instead of the "upload" button.

image

- [ ] Remove current upload button.
- [ ] Replace with big dropzone, the same as in reviewables list.
- [ ] Dropping files/uploading starts upload and opens reviewables tab.

Idea: maybe a React portal from reviewables to the middle section could work?

### PR DESCRIPTION
**Pull request by flynput**
Extracted Drag&Drop & file upload related logic to its own component, using both in reviewablesList and the viewer main pane when no reviewables exist.

https://github.com/user-attachments/assets/6abd5009-c540-4f57-bcef-2581673d559a

Closed #639

### CODE REVIEW

**Comment by flynput** [comment]
@Innders One issue I encountered while testing: viewer context didn't update properly when uploading files on a folder (probably affects other non-product entities as well). I can be fixed of course, but I'm not sure it even makes sense to upload reviewables to folders.

File: src/containers/ReviewablesList/ReviewablesUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+import reviewApi from '@queries/review/getReview'
+import { UploadReviewableApiResponse } from '@api/rest/review'
+import { toggleUpload } from '@state/viewer'
+
+// components
+import ReviewableProgressCard, { ReviewableProgress } from '@components/ReviewableProgressCard'
+import * as Styled from './ReviewablesList.styled'
+import styled from 'styled-components'
+
+export const ReviewablesList = styled.div`
```
Comment (Innders) [inline src/containers/ReviewablesList/ReviewablesUpload.tsx:19]: Worth putting files in a separate files component, not sure, you decide!

File: src/containers/Viewer/Viewer.tsx [hunk 2]
```diff
@@ -288,27 +289,39 @@ const Viewer = ({ onClose }: ViewerProps) => {
         alt={selectedReviewable.label || selectedReviewable.filename}
       />
     )
-  } else if (!isFetchingReviewables) {
+  } else if (!isFetchingReviewables && versionReviewableIds?.length === 0) {
     let message = 'No preview available'
-    let children = null
 
     if (noVersions) {
       message = 'This task has published no versions.'
```
Comment (Innders) [inline src/containers/Viewer/Viewer.tsx:296]: If there are no versions then there is nothing to upload reviewables to so we the upload should be hidden.

Same goes for ''File not supported and needs conversion'.

image

File: src/containers/Viewer/ViewerDialog.tsx [hunk 3]
```diff
@@ -4,16 +4,19 @@ import { closeViewer } from '@state/viewer'
...
 import isHTMLElement from '@helpers/isHTMLElement'
+import clsx from 'clsx'
 
 const StyledDialog = styled(Dialog)`
   /* dnd overlay must offset this 64px by 32px */
   width: calc(100% - 64px);
   height: calc(99% - 64px);
   max-height: unset;
   max-width: unset;
+  &.isHidden {
```
Comment (Innders) [inline src/containers/Viewer/ViewerDialog.tsx:17]: Purpose to this?

File: src/containers/Viewer/ViewerDialog.tsx [hunk 4]
```diff
@@ -4,16 +4,19 @@ import { closeViewer } from '@state/viewer'
 import { useEffect } from 'react'
 import Viewer from './Viewer'
 import styled from 'styled-components'
-import { useLocation } from 'react-router'
 import { $Any } from '@/types'
 import isHTMLElement from '@helpers/isHTMLElement'
+import clsx from 'clsx'
```
Comment (Innders) [inline src/containers/Viewer/ViewerDialog.tsx:9]: Left over import

File: src/containers/ReviewablesList/ReviewablesUpload.tsx
Comment (Innders) [inline src/containers/ReviewablesList/ReviewablesUpload.tsx:1]: Would it be possible to show the drop area when dragging anywhere over the viewer space (where the full dropzone is now) just like how it works with the reviewables list. The placeholder size stays the same but the actual dropzone for files matches what we see when we are dragging.

It's just a bit weird when the full dropzone appears when dragging over the small one.

https://github.com/user-attachments/assets/1a211d9b-cbdf-4e92-b8c3-56f0794f72b9

**Review by Innders — CHANGES_REQUESTED** [review]
Now we can upload reviewables from anywhere in the app, that is awesome!!!

Just a few things to tweak :) Nice one! 😄

File: src/containers/Viewer/ViewerDialog.tsx (hunk 3 above)
Comment (flynput) [reply src/containers/Viewer/ViewerDialog.tsx:17]: Leftover from initial attempts, removing it.

**Comment by flynput** [comment]
@Innders Address all raised issues. Empty placeholder styling changed a bit, particularly keeping the content horizontally centered while on the vertically closer to the top side, let me know if it's close enough to the original.

File: src/containers/Viewer/ViewerComponent.tsx
Comment (Innders) [inline src/containers/Viewer/ViewerComponent.tsx:1]: Not sure if this was a recent change or not but the box can become far too wide. It should be about the width of the text here.

image

**Review by Innders — CHANGES_REQUESTED** [review]

**Pull request merged**